test/unit/startup/test_bootstrap.py

```diff
@@ -536,24 +553,37 @@ def test_existing_worker_successful_restore(
         worker_persistence_info_mock.load.assert_called_once_with(config=config)
         worker_persistence_info_mock.save.assert_not_called()
         create_worker_mock.assert_not_called()
-        worker_info_mock.save.assert_not_called()
+        if get_instance_id_value is not None and saved_instance_id is None:
+            worker_info_mock.save.assert_called_once_with(config=config)
```

Comment:
Do we need to assert that the instance ID was updated?

Reply:
Good catch, updated the test to assert this as well.